public/home.js

```diff
@@ -1,6 +1,99 @@
 $(document).ready(function(){
 
+  var APIaction = function(){
 
+  };
+
+  APIaction.prototype.logOut = function(){
+    $.ajax({
+      type: 'DELETE',
+      url: '/sessions',
+      success: function(response){
+        alert("Logout successful. See you soon!")
+        window.location.href = '/';
+      }
+    })
+  };
+
+  APIaction.prototype.changeDetail = function(){
+    $.ajax({
+      type: 'PUT',
+      url: '/users',
+      data:{
+        user: {
+          firstname: firstName,
+          lastname: lastName,
+          email: email,
+          username: userName,
+          password: password,
+          dateCreated: new Date()
+        }
+      },
+      dataType: 'json',
+      success: function(response){
+        alert("Update successful!")
+        window.location.href = '/home';
+      },
+      error: function(xhr, status, data){
+        console.log(xhr);
+      }
+    })
+  };
+
+  APIaction.prototype.nameDisplay = function(){
+  $.ajax({
+    type: 'GET',
+    url: '/users',
+    dataType: 'json',
+    success: function(response){
+      html  =  '<h4>Welcome Back!<h4>'
+      html +=  '<h4>'+response.firstname.toUpperCase()+" "+response.lastname.toUpperCase()+'</h4>'
+      $('#name_display').replaceWith(html);
+    }
+  });
+}
+
+
+  var apiAction = new APIaction ();
```

Comment:
Also good idea to consider: define the api object definition in another file: `api.js` and then you can keep this file clean to just specify the interactions with jQuery.

It will also allow you to re-use the api in other projects in the future :)